### PR TITLE
Handle test retry for modules

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/helpers/TestDevice.cs [hunk 1]
```diff
@@ -28,7 +28,11 @@ public class TestDevice : IDisposable
     {
         private const int MaxRetryCount = 20;
         private static readonly HashSet<IotHubServiceErrorCode> s_throttlingStatusCodes = new() { IotHubServiceErrorCode.ThrottlingException };
-        private static readonly HashSet<IotHubServiceErrorCode> s_retryableStatusCodes = new(s_throttlingStatusCodes) { IotHubServiceErrorCode.DeviceNotFound };
+        private static readonly HashSet<IotHubServiceErrorCode> s_retryableStatusCodes = new(s_throttlingStatusCodes)
+        {
+            IotHubServiceErrorCode.DeviceNotFound,
+            IotHubServiceErrorCode.ModuleNotFound,
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:34]: Add module not found.

File: e2e/test/helpers/TestDevice.cs [hunk 2]
```diff
@@ -115,26 +119,7 @@ await RetryOperationHelper
...
-                            device = await serviceClient.Devices.GetAsync(requestDevice.Id).ConfigureAwait(false);
-                        }
-                        catch (IotHubServiceException ex)
-                        {
-                            s_logger.Trace($"Getting device {requestDevice.Id} threw {ex}");
-                            device = null;
-                            throw;
-                        }
-
-                        if (device is null)
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:129]: We don't return null anymore.

File: e2e/test/helpers/TestDevice.cs [hunk 3]
```diff
@@ -115,26 +119,7 @@ await RetryOperationHelper
                 .RetryOperationsAsync(
                     async () =>
                     {
-                        try
-                        {
-                            device = await serviceClient.Devices.GetAsync(requestDevice.Id).ConfigureAwait(false);
-                        }
-                        catch (IotHubServiceException ex)
-                        {
-                            s_logger.Trace($"Getting device {requestDevice.Id} threw {ex}");
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:124]: I added this only to test if we always threw or not. No longer needed.

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**